File: codeforces/CodeForcesRound976/b.py
import math
n = int(input())
for _ in range(n):
	a = int(input())
	i = 0
	f = 2e18
	while(i<f):
		m = (i+f)//2
		t = m-math.floor(m**0.5)
		if(t>=a):
			f = m
		else:
			i = m+1
	print(int((i+f)//2))

File: codeforces/CodeForcesRound976/b.cpp
#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main(){
	int n; cin>>n;
	while(n--){
		ll a; cin>>a;
		ll i = 0; ll f = 2e18;
		while(i<f){
			ll m = (i+f)/2;
			ll t = m - floor(sqrt((long double)m));
			if(t>=a) f = m;
			else i = m+1;
		}
		cout<<(i+f)/2<<endl;
	}
}